maas/testing/compare-definitions.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+# Copyright 2015, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import collections
+import json
+import os
+import re
+import sys
+
+
+def load_data(template, directory):
+    path = os.path.join(directory, template)
+    with open(path) as f:
+        data = f.read()
+    return json.loads(data)
+
+
+def compare(reference, new, ignored_keys=None):
```

Comment:
at least for this function, it's confusing to review, variables are somewhat genericly named